plugin-server/tests/postgres/vm.lazy.test.ts

```diff
@@ -126,73 +113,55 @@ describe('LazyPluginVM', () => {
             mocked(createPluginConfigVM).mockImplementation(() => {
-                if (++i === 1) {
-                    throw new Error('I failed without retry, please retry me too!')
-                }
-                throw retryError
+                throw new Error('VM creation failed before setupPlugin')
             })
 
             await vm.initialize!('some log info', 'failure plugin')
-
-            // try to initialize the vm 11 times (1 try + 10 retries)
             await vm.resolveInternalVm
-            for (let i = 0; i < VM_INIT_MAX_RETRIES + 1; ++i) {
-                jest.runOnlyPendingTimers()
-                await vm.resolveInternalVm
 
-                // plugin methods are always null throughout retries
-                expect(await vm.getProcessEvent()).toEqual(null)
-            }
-
-            expect((status.warn as any).mock.calls).toEqual([
-                ['⚠️', 'I failed without retry, please retry me too!'],
-                ['⚠️', 'Failed to load failure plugin. Retrying in 5 s.'],
-                ['⚠️', 'I failed, please retry me!'],
-                ['⚠️', 'Failed to load failure plugin. Retrying in 10 s.'],
-                ['⚠️', 'I failed, please retry me!'],
-                ['⚠️', 'Failed to load failure plugin. Retrying in 20 s.'],
-                ['⚠️', 'I failed, please retry me!'],
-                ['⚠️', 'Failed to load failure plugin. Retrying in 40 s.'],
-                ['⚠️', 'I failed, please retry me!'],
-                [
-                    '⚠️',
-                    'Failed to load failure plugin. Disabling it due to too many retries – tried to load it 5 times before giving up.',
-                ],
-            ])
+            expect((status.warn as any).mock.calls).toEqual([['⚠️', 'VM creation failed before setupPlugin']])
 
             // plugin gets disabled
             expect(disablePlugin).toHaveBeenCalledTimes(1)
             expect(disablePlugin).toHaveBeenCalledWith(mockServer, 39)
         })
 
-        it('vm init will retry on error and load plugin successfully on a retry', async () => {
-            // throw a RetryError setting up the vm
-            mocked(createPluginConfigVM).mockImplementationOnce(() => {
-                throw retryError
+        it('_setupPlugin handles retries correctly', async () => {
+            const mockedRun = jest.fn()
+            const mockVm = {
+                run: mockedRun,
+            }
+            mockedRun.mockImplementation(() => {
+                throw new Error('oh no')
             })
 
-            await vm.initialize!('some log info', 'failure plugin')
-            await vm.resolveInternalVm
+            const lazyVm = createVM()
+
+            await lazyVm._setupPlugin(mockVm as any)
+            await lazyVm._setupPlugin(mockVm as any)
+            await lazyVm._setupPlugin(mockVm as any)
+            await lazyVm._setupPlugin(mockVm as any)
```

Comment:
Use loop over `VM_INIT_MAX_RETRIES` or `max(VM_INIT_MAX_RETRIES, 4)` otherwise if we change the limit to be 3 this test will fail annoyingly